src/Compilers/Core/Portable/Compilation/CompilationExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis
+{
+    public static class CompilationExtensions
+    {
+        /// <summary>
+        /// Gets the low-level operation corresponding to the method's body.
+        /// </summary>
+        /// <param name="compilation">The compilation containing the method symbol.</param>
+        /// <param name="method">The method symbol.</param>
+        /// <param name="cancellationToken">An optional cancellation token.</param>
+        /// <returns>The low-level operation corresponding to the method's body.</returns>
+        public static IOperation GetOperation(this Compilation compilation, IMethodSymbol method, CancellationToken cancellationToken = default)
```

Comment:
>IMethodSymbol method [](start = 76, length = 20)

I think we should require this symbol to be a definition. I.e. should return nothing for a member of a constructed type, or for a constructed generic method. Please add tests.

Reply:
Alternatively we could throw instead of returning null in this case.

---
In reply to: [136207763](https://github.com/dotnet/roslyn/pull/21810#discussion_r136207763) [](ancestors = 136207763)